ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -72,31 +84,58 @@ Output getOutput(Env env) {
 
       @Override
       Output getOutput(Env env) {
-        String gcsPrefix = env.getString(OUTPUT_BUCKET);
-        // Append / to GCS prefix to enforce that it will be a directory
-        if (!gcsPrefix.endsWith("/")) {
-          gcsPrefix += "/";
-        }
-        // Append OUTPUT_TABLE to GCS prefix if present
-        if (env.containsKey(OUTPUT_TABLE)) {
-          gcsPrefix += "output_table=" + env.getString(OUTPUT_TABLE) + "/";
-        }
+        return new Output(env, this, new Gcs.Write.Ndjson(getGcsService(env), //
+            env.getLong(BATCH_MAX_BYTES, 100_000_000L), // default 100MB
+            env.getInt(BATCH_MAX_MESSAGES, 1_000_000), // default 1M messages
+            env.getDuration(BATCH_MAX_DELAY, "10m"), // default 10 minutes
+            PubsubMessageToTemplatedString.of(getGcsOutputBucket(env)), getFormat(env),
+            ignore -> CompletableFuture.completedFuture(null)));
+      }
 
-        // if OUTPUT_TOPIC is present send notification for each gcs blob written
-        final Function<BlobInfo, CompletableFuture<Void>> batchCloseHook;
-        if (env.containsKey(OUTPUT_TOPIC)) {
-          Function<PubsubMessage, CompletableFuture<Void>> pubsubWrite = pubsub.getOutput(env);
-          batchCloseHook = blobInfo -> pubsubWrite.apply(BlobInfoToPubsubMessage.apply(blobInfo));
-        } else {
-          batchCloseHook = ignore -> CompletableFuture.completedFuture(null);
-        }
+      @Override
+      long getDefaultMaxOutstandingElementCount() {
+        return 1_000_000L; // 1M messages
+      }
+
+      @Override
+      long getDefaultMaxOutstandingRequestBytes() {
+        return 1_000_000_000L; // 1GB
+      }
+    },
+    bigQueryLoad {
+
+      @Override
+      Output getOutput(Env env) {
+        return new Output(env, this, new BigQuery.Load(getBigQueryService(env), getGcsService(env),
+            // BigQuery Load API limits maximum bytes per request to 15TB
+            env.getLong(LOAD_MAX_BYTES, 15_000_000_000_000L), // default 15TB
```

Comment:
We patched beam in ingestion-beam to lower this limit, since load jobs start failing on main ping data as the data volume gets to this magnitude. We're using 100 GB as the limit currently.

Though we're unlikely to get anywhere near this volume for streaming jobs.